### DISCUSSION TITLE
unable to use SkAnimatedImage?

### DISCUSSION THREAD

**Discussion by Eywek** [rust-skia/rust-skia#775] [Q&A] [answered]
Hello,

I'm trying to use your amazing project (thank you btw!) to convert my c++ project using Skia to a rust one. 

In c++ I'm currently using `SkAnimatedImage` to decode GIFs:
```c++
#include <include/android/SkAnimatedImage.h>
#include <include/core/SkData.h>
#include <include/codec/SkAndroidCodec.h>

auto bytes = SkData::MakeWithCopy(imageData.data(), imageData.size());
auto aCodec = SkAndroidCodec::MakeFromData(bytes);
auto animatedImage = SkAnimatedImage::Make(aCodec);
animatedImage->setRepetitionCount(0);
animatedImage->decodeNextFrame();
```

But I can't find `AnimatedImage` in this repository, is it something that you could add or maybe is there another way to decode GIF I'm not aware of?

Thank you

**Comment by pragmatrix** [comment]
SkAnimatedImage seems to be specific to Android. You may be able to use SkCodec instead for which skia-safe bindings exist (`Codec`). Skia [supports](https://github.com/rust-skia/rust-skia/tree/master/skia-safe#codecs) loading GIF images, though I don't know if access to individual frames is possible.

**Reply by Eywek** [reply]
Thank you for the reply

It seems that `SkCodec` doesn't export `getFrameInfo` method which prevent me from getting frame duration. I'm also unable to use not exported `SkCodec_kNoFrame` because `codec::Options.prior_frame` is unsigned whereas `SkCodec_Options.fPriorFrame` is a signed int

Any plan on implementing this?

**Comment by Eywek** [comment, marked as answer]
https://github.com/rust-skia/rust-skia/pull/791
